src/mgmt/cli_mgmt.c

```diff
@@ -760,7 +760,8 @@ dc_mgmt_sys_encode(struct dc_mgmt_sys *sys, void *buf, size_t cap)
 		return -DER_TRUNC;
 
 	D_CASSERT(sizeof(sysb->syb_name) == sizeof(sys->sy_name));
-	strncpy(sysb->syb_name, sys->sy_name, sizeof(sysb->syb_name));
+	strncpy(sysb->syb_name, sys->sy_name, DAOS_SYS_NAME_MAX);
+	sysb->syb_name[sizeof(sysb->syb_name) - 1] = '\0';
```

Comment:
Is this simply a Coverity false positive? If sys->sy_name is not null-terminated, something is very broken, and we can't just truncate it here.

Reply:
I considered it to be a false positive too. I believe it is, but do we normally fix it explicitly anyways or ignore in this case?

Rationale behind the change is below:
I saw this code:
src/include/daos/mgmt.h:        char                    sy_name[DAOS_SYS_NAME_MAX + 1];
src/mgmt/cli_mgmt.c:    char    syb_name[DAOS_SYS_NAME_MAX + 1];

In this case if sy_name never gets set to something longer than DAOS_SYS_NAME_MAX then the extra character should be the null termination. While I agree we could leave it as is I wanted to explicitly specify the null termination.

If what I said above is not true, there's a case where it would be above the DAOS_SYS_NAME_MAX value, or if we should just ignore/mark as false positive this Coverity issue please let me know.

Reply:
I would mark the issue as "false positive" and "ignore". Then, there are a few options:

1. Do not make any changes, or
2. Add a D_ASSERT(sysb->syb_name[sizeof(sysb->syb_name) - 1] == '\0').

(sys->sy_name is checked and guaranteed to be null-terminated, unless there's a memory corruption.)